src/v/pandaproxy/schema_registry/sharded_store.cc

```diff
@@ -216,15 +216,12 @@ ss::future<compatibility_level> sharded_store::get_compatibility() {
     co_return _store.local().get_compatibility().value();
 }
 
-ss::future<compatibility_level>
-sharded_store::get_compatibility(const subject& sub) {
-    using overload_t = result<compatibility_level> (store::*)(const subject&)
-      const;
-    auto level = co_await _store.invoke_on(
-      shard_for(sub),
-      _smp_opts,
-      static_cast<overload_t>(&store::get_compatibility),
-      sub);
+ss::future<compatibility_level> sharded_store::get_compatibility(
+  const subject& sub, default_to_global fallback) {
+    auto get = [&sub, &fallback](store& s) {
+        return s.get_compatibility(sub, fallback);
+    };
+    auto level = co_await _store.invoke_on(shard_for(sub), get);
```

Comment:
Are `sub` and `fallback` too big to capture by value? I can't think of any reason why the cross-core reference here would be unsafe, but it does make me stop to think carefully about safety and that's also a reason itself to avoid references if efficiency isn't much of a concern.

Reply:
> Are `sub` and `fallback` too big to capture by value?

I'll change it